### PR TITLE
Replace use of cortex util directory with tempo

### PR DESCRIPTION
**Pull request by annanay25**
Signed-off-by: Annanay <annanayagarwal@gmail.com>

**What this PR does**:
- Replaces the use of cortex util directory by copying relevant files over and using `dskit` wherever necessary
- Shuffle some files around in tempo util directory to better match use
- ⚠️  Drop `PrometheusLogger` -> This no longer tries to register the metric `log_messages_total`. The cortex util package will still register it but it will always show a value of zero. :sadface:

**Which issue(s) this PR fixes**:
Fixes N/A

**Checklist**
- N/A Tests updated
- N/A Documentation added
- N/A `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: pkg/util/http.go [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package util
+
+import (
+	"strings"
+)
+
+// IsRequestBodyTooLarge returns true if the error is "http: request body too large".
+func IsRequestBodyTooLarge(err error) bool {
```
Comment (joe-elliott) [inline pkg/util/http.go:8]: does this belong in /pkg/api?

honestly it only seems to bei n use in `/modules/frontend`. should we just put it there?

File: pkg/util/validation.go [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package util
+
+// Supported values for ingestion rate strategies
+// TODO: check if these can be moved to dskit
+const (
+	LocalIngestionRateStrategy  = "local"
```
Comment (joe-elliott) [inline pkg/util/validation.go:6]: these are already defined in `/modules/overrides`. should we use those?

**Review by joe-elliott — COMMENTED** [review]
Couple o' questions

File: pkg/util/http.go (hunk 1 above)
Comment (annanay25) [reply pkg/util/http.go:8]: makes sense. moved to `frontend/handler.go`

File: pkg/util/validation.go (hunk 2 above)
Comment (annanay25) [reply pkg/util/validation.go:6]: totally missed those, yeah makes sense to use those.

**Review by mapno — APPROVED** [review]

**Pull request merged**